config/application_unit_test.py

```diff
@@ -1,770 +1,768 @@
-    value = simple_search(arg_regex, self.full_arg_string)
-    return is_set(value)
-  #--------------------------------------------------------------------------------------------------#
-
-  #--------------------------------------------------------------------------------------------------#
-  # Checks the list of arguments for arg_name and then check arg_string's
-  # value for check_string
-  def check_arg_value(self, arg_name, check_string):
-    arg_regex = re.compile("{0}=([^\s]*)".format(arg_name))
-    arg_value = simple_search(arg_regex, self.full_arg_string)
-    return ( arg_value.find(check_string) != -1)
-  #--------------------------------------------------------------------------------------------------#
-
-  #--------------------------------------------------------------------------------------------------#
-  # Get argument value from cmake_args dictionary or raise exception
-  # value for check_string
-  def get_arg_value(self, arg_name):
-    return self.cmake_args[arg_name]
-  #--------------------------------------------------------------------------------------------------#
-
-  #--------------------------------------------------------------------------------------------------#
-  # Check to see if test is little endian
-  def is_little_endian(self):
-    return self.little_endian
-  #--------------------------------------------------------------------------------------------------#
-
-#--------------------------------------------------------------------------------------------------#
+# -------------------------------------------------------------------------------------------------#
+
+
+# -------------------------------------------------------------------------------------------------#
+# Class encapsulating all functions needed for unit testing
+class UnitTest:
+    re_app = re.compile(r"APP=([^\s]*)")
+    # re_build_env = re.compile("BUILDENV=([^\s]*)")
+    re_gold_in_file = re.compile(r"GOLDFILE=([^\s]*)")
+    re_std_in_file = re.compile(r"STDINFILE=([^\s]*)")
+    # re_test_args = re.compile("TEST_ARGS=([^\s]*)")
+    # re_pass_regex = re.compile("PASS_REGEX=([^\s]*)")
+    # re_fail_regex = re.compile("FAIL_REGEX=([^\s]*)")
+    # re_pe_list = re.compile("PE_LIST=([^\s]*)")
+    re_project_binary_dir = re.compile(r"PROJECT_BINARY_DIR=([^\s]*)")
+    re_project_source_dir = re.compile(r"PROJECT_SOURCE_DIR=([^\s]*)")
+    re_test_name = re.compile(r"TESTNAME=([^\s]*)")
+    re_numPE = re.compile(r"numPE=([^\s]*)")
+    re_mpiexec = re.compile(r"MPIEXEC_EXECUTABLE=([^\s]*)")
+    re_mpi_cores_per_cpu = re.compile(r"MPI_CORES_PER_CPU=([^\s]*)")
+    re_draco_info = re.compile(r"DRACO_INFO=([^\s]*)")
+    re_gdiff = re.compile(r"GDIFF=([^\s]*)")
+    re_pgdiff = re.compile(r"PGDIFF=([^\s]*)")
+
+    # run command can include spaces, modfiy regular expression for that
+    re_run_cmd = re.compile("RUN_CMD=(.*?)-D")
+
+    re_arg_value = re.compile("ARGVALUE=_av_(.*)_av_")
```

Comment:
* This line is the only design change in this file.
* However, all of these regular expressions needed the `r` (raw string) qualifier to pass flake8 checks.

Reply:
OK, that makes sense. Thanks for the explanation!